test/pkg_autocorrects_test.cc

```diff
@@ -228,23 +264,396 @@ TEST_CASE("Simple add export") {
+    core::GlobalState gs(errorQueue);
+    gs.initEmpty();
+    {
+        core::UnfreezeNameTable packageNS(gs);
+        core::packages::UnfreezePackages unfreezeToEnterPackagerOptionsPackageDB = gs.unfreezePackages();
+        gs.setPackagerOptions({}, {}, {}, {}, {}, {"lib", "app"}, "");
+    }
+
+    string packageWithFakeImport = makePackageRB("HasFakeImport", "false", "app", {"FakeImport"});
+    string packageWithFakeImportPath = "has_fake_import/__package.rb";
+
+    string libPackage = makePackageRB("LibPackage", "false", "lib", {"FalsePackageA"});
+    string libPackagePath = "lib_pkg/__package.rb";
+
+    string appPackage = makePackageRB("AppPackage", "false", "app", {});
+    string appPackagePath = "app_pkg/__package.rb";
+
+    string packageWithComments = "class HasComments < PackageSpec\n"
+                                 "  layer 'app'\n"
+                                 "  strict_dependencies 'false'\n"
+                                 "  import FalsePackageA # a comment\n"
+                                 "end\n";
+    string packageWithCommentsPath = "has_comments/__package.rb";
+
+    auto parsedFiles = enterPackages(gs, {{packageWithFakeImportPath, packageWithFakeImport},
+                                          {falsePackageAPath, falsePackageA},
+                                          {layeredPackageAPath, layeredPackageA},
+                                          {libPackagePath, libPackage},
+                                          {appPackagePath, appPackage},
+                                          {packageWithCommentsPath, packageWithComments}});
+
+    {
+        // Import list contains non-existent package
+        auto &fakeImportPkg = getPackageForFile(gs, parsedFiles[0].file);
+        ENFORCE(fakeImportPkg.exists());
+        auto &layeredPkgA = getPackageForFile(gs, parsedFiles[2].file);
+        ENFORCE(layeredPkgA.exists());
+
+        auto addImport = fakeImportPkg.addImport(gs, layeredPkgA, false);
+        string expected = makePackageRB("HasFakeImport", "false", "app", {"FakeImport", "LayeredPackageA"});
+        ENFORCE(addImport, "Expected to get an autocorrect from `addImport`");
+        auto replaced = addImport->applySingleEditForTesting(packageWithFakeImport);
+        CHECK_EQ(expected, replaced);
+    }
+
+    {
+        // Import added to start of import list
+        auto &libPkg = getPackageForFile(gs, parsedFiles[3].file);
+        ENFORCE(libPkg.exists());
+        auto &appPkg = getPackageForFile(gs, parsedFiles[4].file);
+        ENFORCE(appPkg.exists());
+
+        auto addImport = libPkg.addImport(gs, appPkg, false);
+        string expected = makePackageRB("LibPackage", "false", "lib", {"AppPackage", "FalsePackageA"});
+        ENFORCE(addImport, "Expected to get an autocorrect from `addImport`");
+        auto replaced = addImport->applySingleEditForTesting(libPackage);
+        CHECK_EQ(expected, replaced);
+    }
+
+    {
+        // Import list with comments in it
```

Comment:
This behaviour is technically a bug, but I don't think it's a blocker to merging this PR because:
- the previous version of this code also had this bug (ie. if you had a comment after the last import, the comment would also get moved to the new import)
- at Stripe, we disallow comments in `__package.rb` files between the `class ... end`.